File: wellman-group/backend/app/schemas/certificate.py
from pydantic import BaseModel
from datetime import datetime, date
from uuid import UUID
from typing import Optional


class CertificateCreate(BaseModel):
    title: str
    issuing_body: str
    issue_date: date
    expiry_date: Optional[date] = None
    file_url: str
    order_index: int = 0
    is_active: bool = True


class CertificateUpdate(BaseModel):
    title: Optional[str] = None
    issuing_body: Optional[str] = None
    issue_date: Optional[date] = None
    expiry_date: Optional[date] = None
    file_url: Optional[str] = None
    order_index: Optional[int] = None
    is_active: Optional[bool] = None


class CertificateResponse(BaseModel):
    id: UUID
    title: str
    issuing_body: str
    issue_date: date
    expiry_date: Optional[date] = None
    file_url: str
    order_index: int
    is_active: bool
    created_at: datetime

    model_config = {"from_attributes": True}

File: wellman-group/frontend/types/certificate.ts
export interface Certificate {
  id: string;
  title: string;
  issuing_body: string;
  issue_date: string;
  expiry_date: string | null;
  file_url: string;
  order_index: number;
  is_active: boolean;
  created_at: string;
}

export interface CertificateCreate {
  title: string;
  issuing_body: string;
  issue_date: string;
  expiry_date?: string;
  file_url: string;
  order_index?: number;
  is_active?: boolean;
}

export type CertificateUpdate = Partial<CertificateCreate>;
